src/devices/vdml_usd.cpp

```diff
@@ -25,10 +25,12 @@ std::int32_t list_files_raw(const char* path, char* buffer, int32_t len) {
 	return usd_list_files_raw(path, buffer, len);
 }
 
-std::vector<std::string> list_files(const char* path, char* buffer, int32_t len) {
+std::vector<std::string> list_files(const char* path) {
 	std::vector<std::string> files = {};
+	// malloc buffer to store file names
+	char *buffer = (char *) malloc(10000);
 	// Call the C function
-	int32_t success = usd_list_files_raw(path, buffer, len);
+	int32_t success = usd_list_files_raw(path, buffer, 10000);
```

Comment:
Could you have a `constexpr size_t buffer_size = 10000;` to reuse between these two instances?